main.py

```diff
@@ -141,17 +154,62 @@ def login(driver, username, password):
     email_input = driver.find_element_by_xpath("//input[@name='emailAddress']")
     email_input.clear()
     email_input.send_keys(username)
+    
     password_input = driver.find_element_by_xpath("//input[@name='password']")
     password_input.clear()
     password_input.send_keys(password)
 
     LOGGER.info("Logging in")
     driver.find_element_by_xpath("//input[@value='SIGN IN']").click()
-    wait_until_visible(driver=driver, xpath="//a[@data-path='myAccount:greeting']")
-
+    
+    wait_until_visible(driver=driver, xpath="//a[@data-path='myAccount:greeting']", duration=5)
+    
     LOGGER.info("Successfully logged in")
 
+def retry_login(driver, username, password):
+    sleep_val = 3
+    num_retries_attempted = 0
+    num_retries = 5
+    while True:
+        try:
+            time.sleep(sleep_val)
+            # Xpath to error dialog button
+            xpath = "/html/body/div[2]/div[3]/div[3]/div/div[2]/input"
+            wait_until_visible(driver=driver, xpath=xpath, duration=5)
+            driver.find_element_by_xpath(xpath).click()
+        
+            password_input = driver.find_element_by_xpath("//input[@name='password']")
+            password_input.clear()
+            password_input.send_keys(password)
 
+            LOGGER.info("Logging in")
+            
+            try:
+                driver.find_element_by_xpath("//input[@value='SIGN IN']").click()
+            except Exception as e:
+                if num_retries_attempted < num_retries:
+                    num_retries_attempted += 1
+                    continue
+                else:
+                    LOGGER.info("Too many login attempts. Please restart app.")
+                    break
+                	
+            sleep_val += random.randint(0,5)
+
+            if num_retries_attempted < num_retries:
+                num_retries_attempted += 1
+                continue
+            else:
+                LOGGER.info("Too many login attempts. Please restart app.")
+                break
+        except Exception as e:
+            LOGGER.exception("Error dialog did not load, proceed. Error: " + str(e))
+            break
+
+    wait_until_visible(driver=driver, xpath="//a[@data-path='myAccount:greeting']")
```

Comment:
Everything looks okay to me. Only thing I can think of now would be if too many login attempts occur, we'll still unnecessarily wait 10s for the sign-in greeting to show.

Reply:
Which would be preferable: removing sleep @ line 175 or reducing sleep values (sleep_val = 1 @ line 170, sleep_val += random.randint(0,2) @ line 197? May or may not be ok with these lines modified or removed to decrease wait time. In my testing, I've seen 2 error dialogs appear at the most with current code.